ConsoleApp1/Program.cs

```diff
@@ -1,9 +1,207 @@
+            Console.WriteLine("Select the book to see its full description. 0 - to leave");
+            Library.ListofBooks();
+            index = int.Parse(Console.ReadLine());
+
+            while (index != 0)
+			{
+                Console.Clear();
+                Library.ListofBooks();
+				Console.WriteLine("*************************");
+                Library.books[index - 1].BookInfoFull();
+                Console.WriteLine("*************************");
+                Console.WriteLine("Choose another book to see its full description. 0 - to leave.");
+                index = int.Parse(Console.ReadLine());
+            }
+            Console.Clear();
+            DisplayMenu();
+        }
+
+		/// <summary>
+		/// Delete a reader card from library
+		/// </summary>
+		static void DeleteReaderCard()
 		{
+			Console.Clear();
+            Console.WriteLine("Choose card id to delete. 0 - to leave.");
+            Library.ListofCards();					
+			int index = int.Parse(Console.ReadLine());
+			Library.RemoveCard(--index);
+        }
+
+		/// <summary>
+		/// Add a book from the library to the particular reader card 
+		/// </summary>
+		static void TakeBook()
+		{
+			Console.Clear();
+			Console.WriteLine("Select your card id. 0 - to leave");
+            Library.ListofCards();
+            int id = int.Parse(Console.ReadLine());
+			Console.Clear();
+
+			if (id == 0)
+			{
+				return;
+			}
+
+			Console.WriteLine("0 - Leave");
+			Console.WriteLine("1 - View the full information of this card");
+			Console.WriteLine("2 - Take a new book");
+			Console.WriteLine("3 - Return book to the library");
+            int index = int.Parse(Console.ReadLine());
+
+            switch (index)
+			{
+				case 0:					
+					break;
+				
+				case 1:
+					Console.Clear();
+                    Console.WriteLine("0 - Leave");
+                    Console.WriteLine("1 - Take a new book");
+					Console.WriteLine("2 - Show the list of all read books");
+                    Library.cards[index - 1].LibraryCardFullInfo();
+                    index = int.Parse(Console.ReadLine());
+
+					if (index == 0)
+					{
+						break;
+					}
+					else if (index == 1)
+					{
+                        goto case 2;
+                    }
+					else if (index == 2)
+					{
+                        Console.Clear();
+                        Library.cards[id - 1].AllReadBooks();                        
+                    }	
+                    break;
+				
+				case 2:
+                    Console.WriteLine("What book do you want to take?");
+                    string query = Console.ReadLine();
+                    Library.cards[id - 1].AssignBook(query);					
+                    break;
+				
+				case 3:
+					Library.cards[id - 1].RemoveBook();
+					break;
+            }
+			Console.Clear();
+			DisplayMenu();
+        }
+
+		/// <summary>
+		/// Add a new reader to the library by creading a library card
+		/// </summary>
+		static void NewReader()
```

Comment:
these methods not related to Program class, better to move them to the BookLibrary

Reply:
yes, probably you are right. As a beginner I didn't consider it this way. Thank you, fixed.